### DISCUSSION TITLE
Robot randomly moves the dock on the map

### DISCUSSION THREAD

**Discussion by Einkatzi** [Hypfer/Valetudo#1953] [Q&A / Support] [answered]
First of all, I really love having a cloud-free vacuum robot. If it weren't for Valetudo, I never would have dared to introduce one into my home.

However, I'm having a problem that I think is caused by Valetudo, but I might be wrong.
The robot in case is a Proscenic M6 Pro, running Valetudo 2023.12.0 since today, but the problem appeared under 2023.10.0 as well.
Persistent map is activated and working, and the map is divided into 5 segments.
Right now, almost whenever I start a segment cleaning, the robot will start by leaving the dock, going forward a few cm, making a 360° turn and then alters its position and the docks position on the map, almost at random (often to a completely different room). So of course it won't find the room it's supposed to clean.
I think this is being caused by the starting room looking "different" to the robot compared to the last full mapping pass. The dock is situated under the dining table, so the chairs are almost guaranteed to be in a different position (I have to move them around to allow the robot to get out from under the table). Also, the other objects in the dining room often change places.
So every time I start a segment cleaning a few days after the full mapping pass, the robot finds that the room looks too different and tries to relocate itself and the base to a spot on the map that looks more like what it sees now.

I think it might be thought of as a feature, when you have rather immobile furniture, relocate the dock to a different place and don't want to delete the map and do a full mapping pass just to tell the robots where it is now.
But for me, with rather variable furniture and a fixed spot for the dock, this is an utter nuisance.

The question is: is this behaviour controlled by Valetudo, or by the robots firmware?

And if it's controlled by Valetudo, is it possible to make this behaviour ("relocate the dock on the map if it appears to have moved") optional?

**Comment by stefanmd023** [comment, marked as answer]
all navigation logic is controlled by the robot's firmware

**Comment by Einkatzi** [comment]
That's the answer I was hoping not to get... but still, thank you!
Since this behaviour happens somewhere in the interplay between the robot's firmware and the Valetudo "cloud", it's hard to determine as a user whose fault this is. Seems like the Viomi V6 SLAM algorithm is prone to fail under my circumstances.

If someone else has (had) this problem with the 3irobotix CRL-200S clones running Viomi V6 firmware - have you worked around this somehow? If so, how?

The only idea I have so far is to put the dock on a certain spot where it's possible to put a large object in front of it with just enough room for the robot to get around (or would a light curtain work?*), so the first thing the robot "sees" after leaving the dock for a few centimeters and doing this 360° orientation spin is always the same, even if the furniture in the room has moved significantly.

(* I haven't found specs on the wavelength of the used lidar scanner, but this paper ( https://icosmos.cs.umd.edu/images/2_publication/papers/LidarPhone_SenSys20_nirupam.pdf ) in figure 15 gives information on some materials and whether they're lidar-transparent or not. So anything that's light enough to not register as a "bump" that stops the robot's motion while being lidar-opaque should work. Viomi V6 firmware doesn't implement CollisionAvoidantNavigationControlCapability, so I hope it doesn't try to avoid bumping into the curtain.)

**Reply by stefanmd023** [reply]
afaik all robot manuals mention a minimum clearance they need around the dock. they usually over-estimate it and you can work with less. in your case I would avoid placing the dock in so often a changeable environment. consider somewhere under a sofa, cupboard or kitchen counter (remove the plint)
